fn main() {
    <Self>::foo; //~ ERROR E0411
}
